open_humans/account_views.py

```diff
@@ -56,8 +81,15 @@ def get_template_names(self):
 
         return [self.template_name]
 
+    def post(self, request, *args, **kwargs):
+        ret = super().post(self, request, *args, **kwargs)
+        try:
+            return redirect(unquote_plus(ret.url))
```

Comment:
It get set at line 151 of allauth.account.utils

The problem is the way that I am encoding the return url sort of short circuits django's HttpResponseRedirect, so that is why I added the extra code.  I will comment to that affect.